test/models/compute_resources/ovirt_test.rb

```diff
@@ -13,6 +13,18 @@ class OvirtTest < ActiveSupport::TestCase
     assert_equal host, as_admin { cr.associated_host(vm) }
   end
 
+  describe 'validate' do
+    test 'valid if url is present' do
```

Comment:
i definitely know, that vmware is either http or http. I don't know for the others for sure.